src/dataset.jl

```diff
@@ -172,13 +172,131 @@ end
+* `chunksize` (Integer): Number of features to write in one database transaction. Neglected when `use_gdal_copy` is true. 
+Default is 20000.
+* `strict` (Bool): Set this to `true` if the written dataset should be a 1:1 copy of the source data, default is `false`,
+which allows the driver to adapt if necessary.
+    
+### Returns
+    `nothing`
 """
 function write(
     dataset::AbstractDataset,
     filename::AbstractString;
     kwargs...,
 )::Nothing
-    destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    if nraster(dataset) > 0 && nlayer(dataset) > 0
+        drivername = :driver ∈ keys(kwargs) ? shortname(kwargs[:driver]) : shortname(getdriver(dataset))
+        error("Writing datasets with raster and vector data is not supported when using driver $drivername. 
+        Please file an issue at https://github.com/yeesian/ArchGDAL.jl/issues 
+        including following dataset information: \n\n$dataset")
+    elseif nraster(dataset) > 0
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    elseif nlayer(dataset) > 0
+        writelayers(dataset, filename; kwargs...)
+    end
+    return nothing
+end
+
+"""
+    writelayers(dataset, filename; kwargs...)
+
+Writes the vector dataset to the designated filename. The options are passed to the newly created dataset and
+have to be given as a list of strings in KEY=VALUE format. The chunksize controls the number of features written 
+in each database transaction, e.g. for SQLite. This function can also be used to copy datasets on disk.
+
+Currently working drivers: FlatGeobuf, GeoJSON, GeoJSONSeq, GML, GPKG, JML, KML, MapML, ESRI Shapefile, SQLite
+
+### Parameters
+* `dataset`  The source dataset
+* `filename` The file name to write to
+
+### Keyword arguments
+* `driver`:           The driver to use, you have to manually select the right driver for the file extension you wish
+* `options`:          A vector of strings containing KEY=VALUE pairs for driver-specific creation options
+* `layer_options`:    Driver specific options for layer creation. The options can either be a Vector{String} to provide the
+same options for each layer, or a Vector{Vector{String}} to provide individual options per layer, in the order of their 
+appearance in the dataset. The strings have to be KEY=VALUE pairs. An example for two layers: 
+`[["FORMAT=WKT", "LAUNDER=NO"], ["STRICT=NO"]]`
+* `chunksize`:        Number of features to write in one database transaction. Neglected when `use_gdal_copy` is true.
+* `use_gdal_copy`:    Set this to false (default is true) to achieve higher write speeds at the cost of possible errors. 
+Note that when set to true, no coordinate transformations are possible while writing the features.
+
+### Returns
+nothing
+"""
+function writelayers(dataset::AbstractDataset,
+                     filename;
+                     driver=getdriver(dataset),
+                     options::Vector{String}=[""], 
```

Comment:
nit:
```julia
options = StringList(C_NULL),
```

(It should be untyped and defaulting to NULL for consistency with the rest of the file)

Reply:
I made `options` and `layer_options` untyped, but kept the `[""]` default. Using a `StringList(C_NULL)` breaks the GPKG workaround, because there is no `contains` function for `Ptr{Cstring}` and `String`. Exposing a `Ptr{Cstring}` is maybe also a bit unwieldy for users.

Reply:
Got it, that makes sense, thank you! My (mild and possibly unfounded) concern was that `[""]` might result in different behavior than `StringList(C_NULL)`, but I agree with you that the GPKG workaround is more important here.